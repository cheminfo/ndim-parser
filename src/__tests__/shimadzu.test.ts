import { readFileSync } from 'fs';
import { join } from 'path';

import { test, expect } from 'vitest';

import { ndParse } from '../ndParse';

test('shimadzu test1', () => {
  const blob = readFileSync(join(__dirname, 'data/test1.txt'));
  const result = ndParse(blob, { separator: '\t' });
  expect(Object.keys(result.variables)).toHaveLength(20);
  expect(result).toMatchSnapshot();
});

/*
test.only('shimadzu test2', () => {
  const blob = readFileSync(join(__dirname, 'data/test2.txt'), 'utf8');
  const result = ndParse(blob, { separator: '\t' });
  expect(Object.keys(result.variables)).toHaveLength(20);
  expect(result).toMatchSnapshot();
});
*/

test('shimadzu test3', () => {
  const blob = readFileSync(join(__dirname, 'data/test3.txt'));
  const result = ndParse(blob, { separator: '\t' });
  expect(Object.keys(result.variables)).toHaveLength(2);
  expect(result).toMatchSnapshot();
});
