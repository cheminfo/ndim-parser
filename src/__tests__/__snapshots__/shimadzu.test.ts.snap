// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`shimadzu 1`] = `
{
  "meta": {
    "# of Peaks": "7",
  },
  "variables": {
    "A": {
      "data": [
        4275,
        3094871,
        4656,
        141463,
        268456,
        39166877,
        4499,
      ],
      "label": "Area",
    },
    "C": {
      "data": [
        0,
        0,
        0,
        65.738,
        37.211,
        0,
        0,
      ],
      "label": "Conc.",
    },
    "F": {
      "data": [
        2.455,
        2.865,
        3.323,
        3.962,
        4.186,
        4.788,
        4.727,
      ],
      "label": "F.Time",
    },
    "H": {
      "data": [
        987,
        837061,
        2562,
        69857,
        169781,
        8424638,
        2206,
      ],
      "label": "Height",
    },
    "I": {
      "data": [
        2.31,
        2.571,
        3.22,
        3.799,
        3.962,
        4.186,
        4.634,
      ],
      "label": "I.Time",
    },
    "N": {
      "data": [
        0,
        0,
        0,
        0,
      ],
      "label": "Name",
    },
    "P": {
      "data": [
        5791,
        12172,
        76571,
        87369,
        184181,
        15850,
        115541,
      ],
      "label": "Plate #",
    },
    "R": {
      "data": [
        0,
        2.54,
        8.486,
        12.391,
        4.407,
        2.154,
        4.258,
      ],
      "label": "Resolution",
    },
    "S": {
      "data": [
        0,
        0,
        3.168,
        1.695,
        1.132,
        1.11,
        1.214,
      ],
      "label": "Sep.Factor",
    },
    "T": {
      "data": [
        0,
        1.35,
        1.284,
        1.103,
        1.506,
        1.192,
        1.324,
      ],
      "label": "Tailing",
    },
    "a": {
      "data": [
        4.33,
        3.697,
        1.817,
        2.025,
        1.581,
        4.649,
        2.04,
      ],
      "label": "A/H",
    },
    "b": {
      "data": [
        0,
        0,
        0,
        1,
        2,
        0,
        3,
      ],
      "label": "ID#",
    },
    "c": {
      "data": [
        25.904,
        12.323,
        1.959,
        1.717,
        0.814,
        9.464,
        1.298,
      ],
      "label": "Plate Ht.",
    },
    "d": {
      "data": [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
      ],
      "label": "Area Ratio",
    },
    "e": {
      "data": [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
      ],
      "label": "Height Ratio",
    },
    "f": {
      "data": [
        0,
        0,
        0,
        63.855,
        36.145,
        0,
        0,
      ],
      "label": "Conc. %",
    },
    "g": {
      "data": [
        0,
        0,
        0,
        63.855,
        36.145,
        0,
        0,
      ],
      "label": "Norm Conc.",
    },
    "k": {
      "data": [
        0,
        0.118,
        0.375,
        0.635,
        0.719,
        0.798,
        0.968,
      ],
      "label": "k'",
    },
    "x": {
      "data": [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
      ],
      "label": "Peak#",
    },
    "y": {
      "data": [
        2.368,
        2.648,
        3.255,
        3.871,
        4.07,
        4.256,
        4.661,
      ],
      "label": "R.Time",
    },
  },
}
`;

exports[`shimadzu test1 1`] = `
{
  "meta": {
    "# of Peaks": "7",
  },
  "variables": {
    "A": {
      "data": [
        4275,
        3094871,
        4656,
        141463,
        268456,
        39166877,
        4499,
      ],
      "label": "Area",
    },
    "C": {
      "data": [
        0,
        0,
        0,
        65.738,
        37.211,
        0,
        0,
      ],
      "label": "Conc.",
    },
    "F": {
      "data": [
        2.455,
        2.865,
        3.323,
        3.962,
        4.186,
        4.788,
        4.727,
      ],
      "label": "F.Time",
    },
    "H": {
      "data": [
        987,
        837061,
        2562,
        69857,
        169781,
        8424638,
        2206,
      ],
      "label": "Height",
    },
    "I": {
      "data": [
        2.31,
        2.571,
        3.22,
        3.799,
        3.962,
        4.186,
        4.634,
      ],
      "label": "I.Time",
    },
    "N": {
      "data": [
        0,
        0,
        0,
        0,
      ],
      "label": "Name",
    },
    "P": {
      "data": [
        5791,
        12172,
        76571,
        87369,
        184181,
        15850,
        115541,
      ],
      "label": "Plate #",
    },
    "R": {
      "data": [
        0,
        2.54,
        8.486,
        12.391,
        4.407,
        2.154,
        4.258,
      ],
      "label": "Resolution",
    },
    "S": {
      "data": [
        0,
        0,
        3.168,
        1.695,
        1.132,
        1.11,
        1.214,
      ],
      "label": "Sep.Factor",
    },
    "T": {
      "data": [
        0,
        1.35,
        1.284,
        1.103,
        1.506,
        1.192,
        1.324,
      ],
      "label": "Tailing",
    },
    "a": {
      "data": [
        4.33,
        3.697,
        1.817,
        2.025,
        1.581,
        4.649,
        2.04,
      ],
      "label": "A/H",
    },
    "b": {
      "data": [
        0,
        0,
        0,
        1,
        2,
        0,
        3,
      ],
      "label": "ID#",
    },
    "c": {
      "data": [
        25.904,
        12.323,
        1.959,
        1.717,
        0.814,
        9.464,
        1.298,
      ],
      "label": "Plate Ht.",
    },
    "d": {
      "data": [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
      ],
      "label": "Area Ratio",
    },
    "e": {
      "data": [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
      ],
      "label": "Height Ratio",
    },
    "f": {
      "data": [
        0,
        0,
        0,
        63.855,
        36.145,
        0,
        0,
      ],
      "label": "Conc. %",
    },
    "g": {
      "data": [
        0,
        0,
        0,
        63.855,
        36.145,
        0,
        0,
      ],
      "label": "Norm Conc.",
    },
    "k": {
      "data": [
        0,
        0.118,
        0.375,
        0.635,
        0.719,
        0.798,
        0.968,
      ],
      "label": "k'",
    },
    "x": {
      "data": [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
      ],
      "label": "Peak#",
    },
    "y": {
      "data": [
        2.368,
        2.648,
        3.255,
        3.871,
        4.07,
        4.256,
        4.661,
      ],
      "label": "R.Time",
    },
  },
}
`;

exports[`shimadzu test3 1`] = `
{
  "meta": {
    "# of Points": "7501",
    "End Time(min)": "5.000",
    "Interval(msec)": "40",
    "Start Time(min)": "0.000",
  },
  "variables": {
    "x": {
      "data": [
        0,
        0.00067,
        0.00133,
        0.002,
        0.00267,
        0.00333,
        0.004,
        0.00467,
        0.00533,
        0.006,
        0.00667,
        0.00733,
        0.008,
        0.00867,
        0.00933,
        0.01,
      ],
      "label": "R.Time (min)",
    },
    "y": {
      "data": [
        -185,
        -192,
        -199,
        -205,
        -212,
        -218,
        -225,
        -232,
        -238,
        -244,
        -251,
        -257,
        -264,
        -270,
        -276,
        -283,
      ],
      "label": "Intensity",
    },
  },
}
`;
